volatility3/framework/plugins/windows/pe_symbols.py

```diff
@@ -0,0 +1,777 @@
+            mod_module = PESymbols._get_pdb_module(
+                context, config_path, mod_name, module_info
+            )
+            if mod_module:
+                yield mod_module
+
+    @staticmethod
+    def _find_symbols_through_exports(
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        module_instances: List[Tuple[str, int, int]],
+        mod_name: str,
+    ) -> Generator[ExportSymbolFinder, None, None]:
+        """
+        Attempts to resolve the symbols in `wanted_symbols` through export analysis
+        """
+        pe_table_name = intermed.IntermediateSymbolTable.create(
+            context, config_path, "windows", "pe", class_types=pe.class_types
+        )
+
+        # for each process layer and VAD, construct a PE and examine the export table
+        for module_info in module_instances:
+            exported_symbols = PESymbols._get_exported_symbols(
+                context, pe_table_name, mod_name, module_info
+            )
+            if exported_symbols:
+                yield exported_symbols
+
+    @staticmethod
+    def _get_symbol_value(
+        wanted_modules: PESymbolFinder.cached_value_dict,
+        mod_name: str,
+        symbol_resolver: PESymbolFinder,
+    ) -> Generator[Tuple[str, int], None, None]:
+        """
+        Enumerates the symbols specified as wanted by the calling plugin
+        """
+        wanted_symbols = wanted_modules[mod_name]
+
+        if (
+            PESymbols.wanted_names not in wanted_symbols
+            and PESymbols.wanted_addresses not in wanted_symbols
+        ):
+            vollog.warning(
+                f"Invalid `wanted_symbols` sent to `find_symbols` for module {mod_name}. addresses and names keys both misssing."
+            )
+            return
+
+        symbol_keys = [
+            (PESymbols.wanted_names, "get_address_for_name"),
+            (PESymbols.wanted_addresses, "get_name_for_address"),
+        ]
+
+        for symbol_key, symbol_getter in symbol_keys:
+            # address or name
+            if symbol_key in wanted_symbols:
+                # walk each wanted address or name
+                for wanted_value in wanted_symbols[symbol_key]:
+                    symbol_value = symbol_resolver.__getattribute__(symbol_getter)(
+                        wanted_value
+                    )
+                    if symbol_value:
+                        # yield out symbol name, symbol address
+                        if symbol_key == PESymbols.wanted_names:
+                            yield wanted_value, symbol_value  # type: ignore
+                        else:
+                            yield symbol_value, wanted_value  # type: ignore
+
+                        index = wanted_modules[mod_name][symbol_key].index(wanted_value)  # type: ignore
+
+                        del wanted_modules[mod_name][symbol_key][index]
+
+                # if all names or addresses from a module are found, delete the key
```

Comment:
Sure, and that's fine, but are things accessing `symbols` in between yields of this?  And if so, for what purpose, because it's only going to be partial at that point?